src/main/java/fr/sncf/osrd/infra/railscript/RSExpr.java

```diff
@@ -702,5 +702,41 @@ public void accept(RSExprVisitor visitor) throws InvalidInfraException {
         }
     }
 
+    
+    public static final class PreviousReservedRoute extends RSExpr<RSOptional<RouteState>> {
+        public final RSExpr<SignalState> signal;
+
+        public PreviousReservedRoute(RSExpr<SignalState> signal) {
+            this.signal = signal;
+        }
+
+        /**
+         * @param state
+         * @return the current reserved route preceding the signal, if any. Else an empty optional
+         */
+        @Override
+        public RSOptional<RouteState> evaluate(RSExprState<?> state) {
+            var currentSignal = signal.evaluate(state).signal;
+            var routes = currentSignal.linkedDetector.getIncomingRouteNeighbors(currentSignal.direction.directionSet[0]);
```

Comment:
Yes, it was a clearly the point I wanted to talked about the other day. This is valid if and only if the `Signal` is always a unidirectional signal. I'm not sure that we can give a meaningfull `ApplicableDirection` as parameter. We could give a `Route` but the whole point was to avoid to use the `Route`.